File: backend_python/prompts/interview.py
NEXT_QUESTION_SYSTEM_PROMPT = (
    "你是一个真实、专业、有压迫感但不过度刁难的 AI 面试官。"
    "只输出 JSON，不要输出 Markdown。"
    "JSON 格式必须是 {\"stage\":\"\",\"stability\":\"\",\"focus\":\"\",\"prompt\":\"\"}。"
    "focus 是当前问题的考察点标题，必须贴合具体问题，例如 RAG 召回链路追问、简历真实性核验、FastAPI 模块拆分，最多 16 个中文字符。"
    "你需要根据候选人的简历、岗位 JD、公司要求、历史回答和 RAG 检索上下文生成下一题。"
    "岗位知识库上下文提供岗位知识、可追问方向、评分点和风险信号。"
    "题库 RAG 提供可参考的真实题目和答题要点。"
    "你要吸收这些信息，但不要逐字复述题库原题，除非当前阶段非常适合直接提问。"
    "候选人历史画像提供了过往弱点；如果弱点和当前岗位相关，可以继续深挖。"
    "当前回答和当前 JD 的优先级高于历史记忆。"
    "问题必须具体、短、贴合当前阶段，最多 80 个中文字。"
    "如果上一题回答很空，可以问一个聚焦追问；不要一次问多个大问题。"
)

REPORT_SYSTEM_PROMPT = (
    "你是一个稳定、客观、严谨但偏学习辅导的面试复盘教练。只输出 JSON，不要输出 Markdown。"
    "JSON 格式必须是 {\"score\":数字,\"strengths\":[\"\"],\"risks\":[\"\"],\"actions\":[\"\"],"
    "\"questionReviews\":[{\"index\":1,\"focus\":\"\",\"question\":\"\",\"answerStatus\":\"完整|模糊|不会|跑题\","
    "\"whyAsked\":\"\",\"missingPoints\":[\"\"],\"referenceDirection\":\"\",\"trainingAction\":\"\"}],"
    "\"trainingPlan\":{\"weakTopics\":[{\"focus\":\"\",\"reason\":\"\",\"trainingAction\":\"\"}],"
    "\"nextRoundPriority\":[\"\"],\"practiceQuestions\":[\"\"],\"oneMinuteTemplates\":[\"\"],\"shouldRetry\":布尔值}}。"
    "评分范围 0 到 100。strengths、risks、actions 各输出 3 条，每条不超过 45 个中文字。"
    "questionReviews 必须和用户 answers 一一对应，index 从 1 开始。"
    "answerStatus 只能使用 完整、模糊、不会、跑题 四类。"
    "whyAsked 要解释为什么问这道题，可关联岗位、JD、简历、RAG 上下文或上一轮回答。"
    "missingPoints 最多 3 条，每条不超过 30 个中文字。"
    "referenceDirection 不要写成长篇标准答案，而要给出下次如何组织回答。"
    "trainingAction 必须是一个可执行训练动作，不要写空泛鼓励。"
    "trainingPlan 要把本轮问题转化成下一轮训练计划：weakTopics 写薄弱点，nextRoundPriority 写训练顺序，"
    "practiceQuestions 写可练习的问题，oneMinuteTemplates 写 1 分钟回答模板，shouldRetry 表示是否建议重做本轮。"
    "反馈要具体、可执行，语气偏学习辅导，不要羞辱用户。"
    "评分时必须参考岗位知识库 RAG 的评分点和风险信号、题库 RAG 的答题要点，也要参考候选人画像 RAG 的历史弱点。"
    "当前这一轮答案的表现优先于历史记录。"
)


def build_context_message(title: str, content: str) -> dict[str, str]:
    return {
        "role": "system",
        "content": f"{title}：\n{content}",
    }

File: frontend/src/api/interview.ts
import { apiRequest } from "./client";

export type AgentMode = "coach" | "interview";
export type AgentRuntime = "langgraph_mainline" | "classic" | "shadow" | "langgraph_canary";

export interface RuntimeAuditSummary {
  visibleRuntime?: string;
  fallbackUsed?: boolean;
  fallbackReason?: string;
  qualityGateReasons?: string[];
}

export interface WorkflowTraceItem {
  nodeName?: string;
  node?: string;
  inputSummary?: Record<string, unknown>;
  outputSummary?: Record<string, unknown>;
  fallbackUsed?: boolean;
}

export interface InterviewHistoryItem {
  question: string;
  answer: string;
}

export interface NextQuestionPayload {
  applicationProfileId?: number;
  profile?: Record<string, unknown>;
  history?: InterviewHistoryItem[];
  nextStage?: string;
  agentMode?: AgentMode;
  agentRuntime?: AgentRuntime;
}

export interface NextQuestionResponse {
  stage?: string;
  stability?: string;
  focus?: string;
  prompt?: string;
  question?: string;
  nextQuestion?: string;
  decisionSummary?: string;
  agentDecision?: unknown;
  ragReasons?: string[];
  runtimeAudit?: RuntimeAuditSummary;
  workflowTrace?: WorkflowTraceItem[];
  checkpointSummary?: Record<string, unknown>;
  qualityGate?: Record<string, unknown>;
  fallbackSummary?: { used?: boolean; reason?: string };
}

export interface GenerateReportPayload {
  applicationProfileId?: number | null;
  profile?: Record<string, unknown>;
  answers?: InterviewHistoryItem[];
}

export interface ReportResponse extends Record<string, unknown> {
  score: number;
  strengths: string[];
  risks: string[];
  actions: string[];
  questionReviews?: Record<string, unknown>[];
  trainingPlan?: Record<string, unknown>;
}

export async function nextQuestion(payload: NextQuestionPayload): Promise<NextQuestionResponse> {
  return apiRequest<NextQuestionResponse>("/api/interview/next-question", {
    method: "POST",
    body: JSON.stringify({
      profile: payload.profile || {},
      history: payload.history || [],
      nextStage: payload.nextStage || "",
      agentMode: payload.agentMode || "coach",
      agentRuntime: payload.agentRuntime || "langgraph_mainline",
      applicationProfileId: payload.applicationProfileId
    })
  });
}

export async function generateReport(payload: GenerateReportPayload): Promise<ReportResponse> {
  return apiRequest<ReportResponse>("/api/interview/report", {
    method: "POST",
    body: JSON.stringify({
      applicationProfileId: payload.applicationProfileId,
      profile: payload.profile || {},
      answers: payload.answers || []
    })
  });
}
